File: NNalgorithm/NNalgorithm.py
# coding: UTF-8
import numpy as np
import math

p_syouga = np.array([2, 5])
p_myouga = np.array([-1, 2])

data = np.array([1, 1])

def distance(x, y):
	return sum((x-y)**2)


if distance(p_syouga, data) > distance(p_syouga, data):
	print "syouga"
else:
	print "myouga"

File: NNalgorithm/NNalgorithm.R
p_syouga = c(2, 5)
p_myouga = c(-1, 2)
data = c(1, 1)

# ベクトルの距離を測る
sqrt(sum((p_syouga - data)^2)) # sqrt(17)と結果が一緒なのでOK
sqrt(sum((p_myouga - data)^2))

if (sqrt(sum((p_syouga - data)^2)) > sqrt(sum((p_myouga - data)^2))){
  print("みょうがです", quote=F)
} else {
  print("しょうがです", quote=F)
}
